### PR TITLE
docs: add label documentation

### PR DESCRIPTION
**Pull request by raelga**
How the documentation looks like:

https://github.com/3scale-ops/test-infra/tree/docs/add-label-documentation/hack/label_sync

Closes #8 

/kind documentation
/priority important-longterm
/label size/s
/assign
/hold

Needs #10 to be merged

### CODE REVIEW

File: hack/label_sync/labels.yaml [hunk 1]
```diff
@@ -18,91 +18,244 @@ default:
...
+      description: Categorizes issue or PR as related shared across environments.
+      color: 34828C
+      target: both
+      addedBy: anyone
+    - name: env/dev-eng
+      description: Categorizes issue or PR as related to the dev-env environment.
+      color: C2E0C6
+      target: both
+      addedBy: anyone
+    - name: env/stg-saas
```
Comment (slopezz) [inline hack/label_sync/labels.yaml:216]: Can you add the new label env/pro-saas?

Comment (raelga) [reply hack/label_sync/labels.yaml:216]: As discussed over slack, you will add the new label to validate the documentation and current procedure.

**Pull request merged**